test/unit/core/algorithm/configuration_test.cpp

```diff
@@ -60,6 +60,25 @@ TEST(configuration, size)
     EXPECT_EQ(seqan3::configuration{}.size(), 0u);
 }
 
+TEST(configuration, pipe_operator)
+{
+    { // config | configuration_element
+        seqan3::configuration<foobar<>> cfg{};
+        auto combined_cfg = cfg | bar{1};
+
+        EXPECT_EQ(combined_cfg.size(), 2u);
+        EXPECT_TRUE((std::same_as<decltype(combined_cfg), seqan3::configuration<foobar<>, bar>>));
+
+    }
+
+    { // configuration_element | configuration_element
+        auto combined_cfg = bax{} | bar{1};
+
+        EXPECT_EQ(combined_cfg.size(), 2u);
+        EXPECT_TRUE((std::same_as<decltype(combined_cfg), seqan3::configuration<bax, bar>>));
+    }
+}
```

Comment:
If the commit `[TEST] Add configuration pipe operator test.` is before `[FIX] Add pipe operator overload for config_elem | config.`, please change the order with first fixing and then testing. Or merge them. Otherwise, we have commits that fail to build, and we should try that every commit is buildable.

Reply:
All the combinations of rvalue / lvalue are missing

Reply:
We already have a test place for these:

test/unit/core/algorithm/pipeable_config_element_test.cpp

Reply:
> We already have a test place for these:

> test/unit/core/algorithm/pipeable_config_element_test.cpp

Thanks, I was wondering why we don't have them but I didn't look there. I assumed that it should be in the configuration_test since it is a configuration feature. I'll change the tests.